cvmfs/export_plugin/fs_traversal_interface.h

```diff
@@ -0,0 +1,346 @@
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] path The path where the file should be linked
+   * @param[in] identifier The identifier of the file
+   * (usually obtained through get_identifier)
+   */
+  int (*do_link)(struct fs_traversal_context *ctx,
+                const char *path,
+                const char *identifier);
+
+  /**
+   * Method removes the hardlink at the given path
+   * 
+   * Error if:
+   * - Link does not exist
+   * - Link is directory
+   * - Unlink not successful
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] path The path which should be removed
+   */
+  int (*do_unlink)(struct fs_traversal_context *ctx,
+                const char *path);
+
+  /**
+   * Method which will create the given directory
+   * 
+   * If the path already exists the old entry will silently be removed
+   * and a new directory will be created
+   * 
+   * Error if:
+   * - Parent directory not defined
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] path The path to the directory that should be created
+   * @param[in] stat The stat containing the meta data for directory creation
+   */
+  int (*do_mkdir)(struct fs_traversal_context *ctx,
+                const char *path,
+                const struct cvmfs_attr *stat);
+
+  /**
+   * Method which removes the given directory
+   * 
+   * Error if:
+   * - Directory doesn't exist
+   * - Directory is not empty
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] path The path to the directory that should be removed
+   */
+  int (*do_rmdir)(struct fs_traversal_context *ctx,
+                const char *path);
+
+  /**
+   * Atomically creates the file representing
+   * the given identifier
+   * 
+   * Error:
+   * - If file exists (errno set to EEXIST)
+   * - Error during creation and meta data saving operations
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] stat The stat containing the meta data for file creation
+   */
+  int (*touch)(struct fs_traversal_context *ctx,
+                const struct cvmfs_attr *stat);
+
+  /**
+   * Retrieves a handle struct which allows the manipulation of the file
+   * defined by the given identifier
+   * 
+   * @param[in] ctx The file system traversal context
+   * @param[in] identifier The identifier of the file to manipulate
+   * (usually obtained through get_identifier)
+   * @returns A context that can be used for reading from and writing to the
+   * specified file (needs to be freed by ffree method)
+   */
+  void *(*get_handle)(struct fs_traversal_context *ctx,
+                const char *identifier);
+
+
+  /**
+   * Method which creates a symlink at src which points to dest
+   * 
+   * If src already exists the old entry will silently be removed
+   * and a new symlink will be created
+   * 
+   * Error:
+   * - src directory doesn't exist
+   * - symlink creation fails
```

Comment:
If the target is the empty string this fails, which can occur in CVMFS.

Reply:
ok